File: app/src/main/java/igrek/songbook/songselection/search/SongSearchItem.java
package igrek.songbook.songselection.search;

import igrek.songbook.persistence.songsdb.Song;
import igrek.songbook.songselection.tree.SongTreeItem;

public class SongSearchItem extends SongTreeItem {
	
	private SongSearchItem(Song song) {
		super(song, null);
	}
	
	public static SongSearchItem song(Song song) {
		return new SongSearchItem(song);
	}
	
}

File: app/src/main/java/igrek/songbook/songselection/search/SongSearchItem.kt
package igrek.songbook.songselection.search

import igrek.songbook.persistence.general.model.Song
import igrek.songbook.songselection.listview.items.SongTreeItem

class SongSearchItem private constructor(song: Song) : SongTreeItem(song, null) {

    companion object {
        fun song(song: Song): SongSearchItem {
            return SongSearchItem(song)
        }
    }

}
